internal/storage/mock/testify_mock_bucket.go

```diff
@@ -124,3 +125,11 @@ func (m *TestifyMockBucket) CreateFolder(ctx context.Context, folderName string)
 	}
 	return nil, args.Error(1)
 }
+
+func (m *TestifyMockBucket) SyncObject(ctx context.Context, fileName string, srcObject *gcs.Object, content gcsx.TempFile) (o *gcs.Object, err error) {
```

Comment:
MockBucket is for mocking bucket operations. Why are we adding custom operations to this file?

Reply:
Added this so that mock bucket also implements SyncerBucketInterface. I want to use it for file_test.go file which uses syncer bucket.